code/modules/mod/modules/modules_science.dm

```diff
@@ -88,21 +88,33 @@
 	icon_state = "teleporter"
 	module_type = MODULE_ACTIVE
 	complexity = 3
-	use_energy_cost = DEFAULT_CHARGE_DRAIN * 5
-	cooldown_time = 5 SECONDS
+	use_energy_cost = DEFAULT_CHARGE_DRAIN * 25
+	cooldown_time = 4 SECONDS
 	accepted_anomalies = list(/obj/item/assembly/signaler/anomaly/bluespace)
 	required_slots = list(ITEM_SLOT_BACK|ITEM_SLOT_BELT)
 	/// Time it takes to teleport
-	var/teleport_time = 3 SECONDS
+	var/teleport_time = 1 SECONDS
+	/// Maximum turf range
+	var/max_range = 9
 
 /obj/item/mod/module/anomaly_locked/teleporter/on_select_use(atom/target)
 	. = ..()
 	if(!.)
 		return
 	var/turf/open/target_turf = get_turf(target)
-	if(!istype(target_turf) || target_turf.is_blocked_turf_ignore_climbable() || !(target_turf in view(mod.wearer)))
+	if(get_dist(target_turf, mod.wearer) > max_range)
```

Comment:
how does this interact with mesons?

Reply:
I'm more concerned if this allows you to teleport into blackness

Reply:
tried and no

Reply:
bundled a LoS refactor to handle mesons